src/components/application_manager/include/application_manager/application.h

```diff
@@ -188,9 +196,16 @@ class DynamicApplicationData {
   virtual const smart_objects::SmartObject* keyboard_props() const = 0;
   virtual const smart_objects::SmartObject* menu_title() const = 0;
   virtual const smart_objects::SmartObject* menu_icon() const = 0;
-  virtual const smart_objects::SmartObject* day_color_scheme() const = 0;
-  virtual const smart_objects::SmartObject* night_color_scheme() const = 0;
-  virtual const std::string& display_layout() const = 0;
+  virtual smart_objects::SmartObject day_color_scheme() const = 0;
+  virtual smart_objects::SmartObject night_color_scheme() const = 0;
+  virtual std::string display_layout() const = 0;
```

Comment:
Why change these from pointers and references?

Reply:
`window_templates_` map was introduced in this feature, and we need to perform search for default window to access color schemes. Handling negative search cases is simpler with constructing objects. Given that null objects are safer than null ptrs and these smart objects are relatively small, memory overhead would not be so great.

Reply:
@JackLivio to add to the comment above - As well as there is no need to track that all dynamically allocated memory will be freed in the constructor as it was done before.

Reply:
👍